borg/archiver.py

```diff
@@ -1488,6 +1488,10 @@ def sig_info_handler(signum, stack):  # pragma: no cover
             break
 
 
+def sig_term_handler(signum, stack):
+    raise KeyboardInterrupt()
```

Comment:
Other than the name "KeyboardInterrupt" being not entirely accurate I don't see why this would be dirty - it's the same thing Python does for SIGINT raising KeyboardInterrupt (just in C and some more complexity to defer stuff correctly to the Python VM etc)